docs/docsite/rst/testing_modules.rst

```diff
@@ -0,0 +1,196 @@
+  aci_username=<apic-username> aci_password= <apic-password>
+
+2. Go to **ansible-aci** in the terminal and test the new module using the following commands. To make it easier to run all the commands in one go, we store the commands in a script and run the script.
+
+.. code-block:: Blocks
+
+      rm -rf cisco-aci-*
+      ansible-galaxy collection build --force
+      ansible-galaxy collection install cisco-aci-* --force
+      cd ~/.ansible/collections/ansible_collections/cisco/aci
+      ansible-test sanity --docker --color --truncate 0 -v
+      ansible-test network-integration --docker --color --truncate 0 -vvv --coverage aci_<your module name>
+      ansible-test coverage report
+      ansible-test coverage html
+      open ~/.ansible/collections/ansible_collections/cisco/aci/tests/output/reports/coverage/index.html
+
+
+* ansible-galaxy collection build --force builds a collection artifact that can be stored in a central repository. By default, this command builds from the current working directory, which in our case is ansible-aci.
+
+* ansible-galaxy collection install cisco-aci-* --force installs the built collection in our current working directory, ansible-aci.
+
+* cd ~/.ansible/collections/ansible_collections/cisco/aci changes our directory to aci, where tests are performed.
+
+* ansible-test sanity --docker --color --truncate 0 -v is used to run sanity tests inside Docker, which already has all the dependencies.
+
+* ansible-test network-integration --docker --color --truncate 0 -vvv --coverage aci_<your module name> is used to run integration tests inside Docker. We can either run the integration test on one module or all the modules by omitting the name altogether.
+
+* We add the --coverage option to any test command to collect code coverage data:
+    1. ansible-test coverage report
+    2. ansible-test coverage html
+    3. open ~/.ansible/collections/ansible_collections/cisco/aci/tests/output/reports/coverage/index.html
+
+    + In the Ansible test coverage report generated by commands like ansible-test coverage report and ansible-test coverage html, the colors green, yellow, and red indicate the extent of code coverage for your modules:
+
+        + Green: Represents well-covered code sections where tests have executed the code paths. This indicates good test coverage and confidence in the tested code.
+        + Yellow: Indicates partial coverage or code that is executed but not fully tested. This suggests areas where tests could be improved or expanded.
+        + Red: Marks code that is not covered by any tests. These are gaps in testing that could lead to undetected bugs or issues.
+
+    .. note:: A common best practice is to aim for 100% coverage to ensure sufficient testing. While high coverage is desirable, 100% coverage is often not achievable due to untestable corner cases or environment constraints. It is important to still strive for coverage higher than 95% and to document any known gaps in testing.
```

Comment:
think you might want to highlight the exception of if main() call in `if __name__ == "__main__"`